vespalog/src/test/log_message/log_message_test.cpp

```diff
@@ -0,0 +1,254 @@
+// Copyright 2019 Oath Inc. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <gtest/gtest.h>
+#include <vespa/log/log_message.h>
+#include <vespa/log/exceptions.h>
+
+using LogLevel = ns_log::Logger::LogLevel;
+
+namespace ns_log {
+
+namespace {
+
+std::ostream &
+add_time(std::ostream &os, double now)
+{
+    constexpr int64_t onegig = 1000000000;
+    int64_t now64 = now * onegig;
+    int64_t now64i = now64 / onegig;
+    int64_t now64r = now64 % onegig;
+    os << now64i << '.' << std::setw(6) << std::setfill('0') << (now64r / 1000) <<
+        std::setw(0) << std::setfill(' ');
+    return os;
+}
+
+std::string
+build_logline(double now, std::string hostname, std::string pidString, std::string service, std::string component, std::string level, std::string payload)
+{
+    std::ostringstream os;
+    add_time(os, now) << '\t' << hostname << '\t' << pidString << '\t' <<
+        service << '\t' << component << '\t' << level << '\t' << payload;
+    return os.str();
+}
+
+std::string
+build_logline(double now, std::string remaining)
+{
+    std::ostringstream os;
+    add_time(os, now) << '\t' << remaining;
+    return os.str();
+}
+
+}
+
+class LogMessageTest : public ::testing::Test {
+protected:
+    double _now;
+public:
+    LogMessageTest()
+        : _now(0.0)
+    {
+        _now = time(nullptr);
+    }
+    ~LogMessageTest() { }
+};
+
+TEST_F(LogMessageTest, require_that_plain_entry_is_ok)
+{
+    std::string log_line = build_logline(_now, "localhost", "10/20", "test", "testrunner", "warning", "hello world");
```

Comment:
With the removal of "too old" / "too new" checks I think we can simplify this by adding an explicit time stamp instead of now.

Reply:
Tests are rewritten to use explicit time stamp.